Implement DPP toggle_cce

apahl-cl opened the pull request:
As described in EasyMesh R6, Section 5.3.1, proxy agents must be able to advertise the CCE IE in their Beacon and Probe Response frames. The main addition of this PR is `em_t::toggle_cce(bool enable)`, which adds or removes DPP IEs to BSSs to advertise the CCE IE.

Currently, the IEs are added to every BSS found in `em_t::m_data_model`. 

I considered splitting `toggle_cce` into 2 separate functions (`enable_cce` and `disable_cce`), but I decided to stick with `toggle_cce` as it integrates more smoothly into existing code. 

Requesting review from @bcarlson-dev.

bcarlson-dev commented on inc/em_mgr.h:
```diff
@@ -61,6 +61,18 @@ class em_mgr_t {
+     */
+    virtual int refresh_onewifi_subdoc(const char *log_name, const webconfig_subdoc_type_t type) {
+        printf("refresh_onewifi_subdoc not implemented\n");
+        return 0;
```
Can we add a new return value here (`-2`?) and in the docstring for this default situation where it is not implemented?

This is important because both the controller and the agent inheirit from the manager but only the agent can (since it's the only one with direct access to OneWifi) implement this. So it's very possible that someone could accidentially call this on the controllers side which would be very improper.

apahl-cl commented:
Implemented Ben's feedback. Here are all the changes since his review:
- Moved `CCE_IE` into `dm_bss.h`
- Add `toggle_cce` into the `ec_manager_t` constructor
- Remove `toggle_cce` from `ec_manager_t::upgrade_to_onboarded_proxy_agent` function parameters
- Add `toggle_cce` into the `ec_pa_configurator` constructor
- Removed `m_toggle_cce` from `ec_configurator.h` as `ec_pa_configurator` is the only subclass that should be able to toggle CCE
- Add a fourth exit code to `em_agent_t::refresh_onewifi_subdoc`: `-2` means the function is unimplemented
- Use MACSTRFMT as suggested for printing BSSIDs
- Add basic error handling on vendor IE refresh in `em_t::toggle_cce`

apahl-cl commented on src/em/em.cpp:
```diff
@@ -635,6 +635,50 @@ bool em_t::is_matching_freq_band(em_freq_band_t *band)
+
+    // Refresh OneWifi
+    webconfig_subdoc_type_t vap_type = dm_easy_mesh_t::get_subdoc_vap_type_for_freq(get_band());
+    m_mgr->refresh_onewifi_subdoc("'Vendor IE Refresh'", vap_type);
```
My error handling is a bit rudimentary for this, I just check for an unsuccessful return value, print the return value if it's an error, and return false. But I think that's sufficient for `toggle_cce` since it just returns a bool, unless we want to implement finer-grained error return from `toggle_cce`.

bcarlson-dev commented on inc/ec_manager.h:
```diff
@@ -95,10 +97,11 @@ class ec_manager_t {
      * 
      * @param toggle_cce The function to call to toggle CCE presence
      * @return bool true if successful, false otherwise
+     * @note If the operation fails, all CCE IEs are removed before the function exits
```
Can you remove the `@param` from the docstring now that it's no longer needed?

apahl-cl replied:
Of course, done!

bcarlson-dev reviewed (changes requested):
Since that small docstring change is addressed it LGTM!
